scripts/run_workflow.pl

```diff
@@ -93,3 +103,31 @@ sub run {
   my $error = system($cmd);
   if ($error) { exit($error); }
 }
+
+sub get_aliquot_id_from_bam {
+  my $bam = shift;
+  die "BAM file does not exist: $bam" unless ( -e $bam );
+
+  my $command = sprintf q{samtools view -H %s | grep '^@RG'}, $bam;
+  my ($stdout, $stderr, $exit) = capture { system($command); };
+  die "STDOUT: $stdout\n\nSTDERR: $stderr\n" if ( $exit != 0 );
+
+  my %names;
+  for ( split "\n", $stdout ) {
+    chomp $_;
+    if ( $_ =~ m/\tSM:([^\t]+)/ ) {
+      $names{$1} = 1;
+    }
+    else {
+      die "Found RG line with no SM field: $_\n\tfrom: $bam\n";
+    }
+  }
+
+  my @keys = keys %names;
+  die "Multiple different SM entries: ."
+    . join( q{,}, @keys )
+    . "\n\tfrom: $bam\n"
+    if ( scalar @keys > 1 );
+  die "No SM entry found in: $bam\n" if ( scalar @keys == 0 );
+  return $keys[0];
```

Comment:
same commend as the other project, please make sure non `[a-zA-Z0-9_-]` characters in `SM` are converted to `_` before returning it